Lib/test/test_asyncio/test_taskgroups.py

```diff
@@ -1041,6 +1041,50 @@ class MyKeyboardInterrupt(KeyboardInterrupt):
         self.assertListEqual(gc.get_referrers(exc), no_other_refs())
 
 
+    async def test_cancels_task_if_created_during_creation(self):
+        # regression test for gh-128550
+        ran = False
+        class MyError(Exception):
+            pass
+
+        exc = None
+        try:
+            async with asyncio.TaskGroup() as tg:
+                async def third_task():
+                    raise MyError("third task failed")
+
+                async def second_task():
+                    nonlocal ran
+                    tg.create_task(third_task())
+                    with self.assertRaises(asyncio.CancelledError):
+                        await asyncio.sleep(0)  # eager tasks cancel here
+                        await asyncio.sleep(0)  # lazy tasks cancel here
+                    ran = True
+
+                tg.create_task(second_task())
+        except* MyError as excs:
+            exc = excs.exceptions[0]
+
+        self.assertTrue(ran)
+        self.assertIsInstance(exc, MyError)
+
+
+    async def test_cancellation_does_not_leak_out_of_tg(self):
+        class MyError(Exception):
+            pass
+
+        async def throw_error():
+            raise MyError
+
+        try:
+            async with asyncio.TaskGroup() as tg:
+                tg.create_task(throw_error())
+        except* MyError:
+            pass
+
+        await asyncio.sleep(0)
```

Comment:
Why is this needed?

Reply:
it's needed so there's somewhere for the cancellation to cancel if it leaks, the error message is more confusing as the cancellation happens in unittest internals otherwise

Reply:
Add a comment, otherwise someone would remove it without knowing.

Reply:
```suggestion
        # if this test fails this current task will be cancelled
        # outside the TG and inside unittest internals
        # we add somewhere to cancel here so the test failure message
        # is more understandable
        await asyncio.sleep(0)
```